src/sortStudents.ts

```diff
@@ -1,16 +1,74 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desk';
 
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  const result: Student[] = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function averageGrade(student: Student): number {
+    return student.grades.reduce((a: number, b: number) => (
+      a + b), 0) / student.grades.length;
+  }
+
+  switch (sortBy) {
+    case (SortType.Age):
+      if (order === 'asc') {
+        result.sort((a: Student, b: Student): number => a.age - b.age);
+      } else {
+        result.sort((a: Student, b: Student): number => b.age - a.age);
+      }
+      break;
+
+    case (SortType.Married):
+      if (order === 'asc') {
+        result.sort((a: Student, b: Student): number => (
```

Comment:
Тут теж не обов`язково писати тип)

```suggestion
        result.sort((a, b): number => (
```